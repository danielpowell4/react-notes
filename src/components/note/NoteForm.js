import React from 'react'

export const NoteForm = (props) => (
  <form onSubmit={props.handleSubmit}>
    <input type="text"
      onChange={props.handleInputChange}
      value={props.currentNote}/>
  </form>)

  NoteForm.propTypes = {
    currentNote: React.PropTypes.string.isRequired,
    handleInputChange: React.PropTypes.func.isRequired,
    handleSubmit: React.PropTypes.func.isRequired
  }
