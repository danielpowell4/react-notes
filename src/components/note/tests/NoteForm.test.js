import React from 'react';
import renderer from 'react-test-renderer';
import {NoteForm} from '../NoteForm';

/* NoteForm snapshot test */

const noteFormSampleProps = { currentNote: 'new note',
                              handleInputChange: function(){},
                              handleSubmit: function(){} }

test('NoteForm renders correctly', () => {
  const tree = renderer.create(
    <NoteForm {...noteFormSampleProps} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
