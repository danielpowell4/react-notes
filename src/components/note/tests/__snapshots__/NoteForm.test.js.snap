exports[`test NoteForm renders correctly 1`] = `
<form
  onSubmit={[Function]}>
  <input
    onChange={[Function]}
    type="text"
    value="new note" />
</form>
`;
